### ISSUE TITLE
docker: support per container restart policies

### ISSUE THREAD

**Issue by seschwar** [ansible/ansible-modules-core#353]
##### Issue Type:

Feature Idea
##### Ansible Version:

```
% ansible --version
ansible 1.7.2
```
##### Environment:

Debian unstable

```
% cat /etc/debian_version
jessie/sid
% uname -v
#1 SMP Debian 3.16.7-2 (2014-11-06)
% docker --version
Docker version 1.3.1, build 4e9bbfa
% pip show docker-py

---
Name: docker-py
Version: 0.5.3
Location: /usr/lib/python2.7/dist-packages
Requires: requests, six, websocket-client
% python --version
Python 2.7.8
```
##### Summary:

Docker supports per container restart policies since [version
1.2.0](http://blog.docker.com/2014/08/announcing-docker-1-2-0/).
However the current Ansible `docker` modules does not.

I would like this flag to be supported since the global
`--restart` is deprecated and i still want my containers to be
restarted after a reboot in the future.
##### Steps To Reproduce:

Launch a simple container and look at the `failed` and
`RestartPolicy.Name` fields using the following command:

```
ansible -m docker -a 'image=busybox command=true restart=on-failure' localhost | sed '1c{' | jq -c '[.failed, .ansible_facts.docker_containers[0].HostConfig.RestartPolicy.Name]'
```
##### Expected Results:

The container should be created with a restart policy like the
commands `docker run --restart=...` or `docker create
--restart=...` would.  Therefore the output of the command above
should be:

```
[false,"on-restart"]
```
##### Actual Results:

The parameter is not supported and i get a `FAILED` result with
the message "unsupported parameter for module: restart".  The
actual output of the command above is:

```
[true,null]
```

**Comment by dkerwin** [comment]
PR #199 to the rescue

**Comment by bcoca** [comment]
closing this as duplicate of #199 which also implements it

**Comment by seschwar** [comment]
Oh, I missed that. Very nice!
